plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/move/MoveTypeRefactoringParticipant.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Mario Kastner
+ *     - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+package org.eclipse.fordiac.ide.typemanagement.refactoring.move;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.IdentifierVerifier;
+import org.eclipse.fordiac.ide.model.helpers.PackageNameHelper;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.model.search.types.BlockTypeInstanceSearch;
+import org.eclipse.fordiac.ide.model.search.types.DataTypeInstanceSearch;
+import org.eclipse.fordiac.ide.model.typelibrary.DataTypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibraryManager;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.UpdateFBInstanceChange;
+import org.eclipse.fordiac.ide.typemanagement.wizards.Messages;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.CompositeChange;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.participants.CheckConditionsContext;
+import org.eclipse.ltk.core.refactoring.participants.MoveParticipant;
+
+public class MoveTypeRefactoringParticipant extends MoveParticipant {
+
+	private String oldPackageName;
+	private String newPackageName;
+	private TypeEntry type;
+	private IResource destinationFolder;
+
+	@Override
+	protected boolean initialize(final Object element) {
+		if (element instanceof final IFile file) {
+			this.type = TypeLibraryManager.INSTANCE.getTypeEntryForFile(file);
+			this.oldPackageName = type.getPackageName();
+			this.newPackageName = getPackageNameProposal();
+			if (getArguments().getDestination() instanceof final IResource folder) {
+				this.destinationFolder = folder;
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public String getName() {
+		final StringBuilder sb = new StringBuilder(Messages.MoveTypeToPackage_RenamePackageTo);
+		sb.append(": "); //$NON-NLS-1$
+		sb.append(newPackageName);
+		return sb.toString();
```

Comment:
Please use `MessageFormat` to format messages (also need to adapt the message accordingly):
```suggestion
		return MessageFormat.format(Messages.MoveTypeToPackage_RenamePackageTo, newPackageName);
```